src/components/SigningModals/MultiScreenModal.js

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useState } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Modal,
+  Viewport,
+  springs,
+  textStyle,
+  useViewport,
+  GU,
+} from '@aragon/ui'
+import { Spring, Transition, animated } from 'react-spring'
+import { useSteps } from '../../hooks'
+
+const AnimatedDiv = animated.div
+
+const DEFAULT_MODAL_WIDTH = 700
+
+function MultiScreenModal({ visible, screens, onClose }) {
+  const steps = screens.length
+  const { prev, step, next, direction } = useSteps(steps)
```

Comment:
Just curious, but what happens here if the number of steps changes dynamically (e.g. as a result of the `screens` array changing size)?

Reply:
Great question! it currently resets to `0` per `useSteps` implementation:

https://github.com/aragon/aragon/blob/232dc29dfdcc186b929ca1f553cba8aba95eea1c/src/hooks.js#L89-L92

We will probably be dynamically updating depending on the flow, for this reason I think it would be better to remain on the current `step` if length is increased, and use last in array if length is decreased below current index. What do you think?